### PR TITLE
Fixed: for #5162, The beginning of a TextNode with canInsertTextBefore false in …

### PR DESCRIPTION
**Pull request by matsuyama-k1**
Fix for #5162

### CODE REVIEW

**Comment by matsuyama-k1** [comment]
When multiple strings are selected from the beginning of a TextNode with $canInsertTextBefore false and input in languages where composing is performed, such as Japanese and Korean, the text input is not reflected and the editor will not accept input until enter is pressed.
I addressed this problem.
Perhaps the cause of the problem is that in languages where composing occurs, COMPOSITION_START_CHAR is entered by default, so two characters are input.
In my understanding, the onInput function is executed when a node is created and the second character is input, but in Japanese and other languages, it works when a single character is input.
Therefore, canInsertTextBefore malfunctions.
However, it was not possible to directly prevent the problem when two characters are input, so I added an if statement condition to solve the most recent problem.

File: packages/lexical/src/LexicalUtils.ts [hunk 1]
```diff
@@ -650,7 +650,8 @@ export function $updateTextNodeFromDOMContent(
             prevSelection.anchor.offset === 0) ||
             (prevSelection.anchor.key === textNode.__key &&
               prevSelection.anchor.offset === 0 &&
-              !node.canInsertTextBefore()) ||
+              !node.canInsertTextBefore() &&
+              !isComposing) ||
```
Comment (zurfyx) [inline packages/lexical/src/LexicalUtils.ts:654]: Can we do the same on L657 for `canInsertTextAfter`?

**Review by zurfyx — APPROVED** [review]
Thanks!

From a correctness perspective this PR is not great, the `canInsertBefore` was meant to prevent any input at boundaries which is what you're doing here.

Unfortunately, composition is not flexible, and we can't reposition the cursor when that happens, and with the MutationObserver logic combined is the reason why you're not seeing any text.

I believe the best solution would be to swap DOM nodes but that is something we don't support now and might get tricky with custom extended nodes.

Given that this change improves a11y, I'm inclined to accept it despite the correctness trade-off.

File: packages/lexical/src/LexicalUtils.ts (hunk 1 above)
Comment (matsuyama-k1) [reply packages/lexical/src/LexicalUtils.ts:654]: I have inserted the line within the `canInsertTextAfter` context and confirmed that the jest tests run correctly. Additionally, I attempted to run `the npm run test-e2e-chromium` command for end-to-end testing, but the execution time was excessively long, preventing the tests from completing. I haven't encountered any specific scenarios that cause bugs when the line is not inserted within canInsertTextAfter, but it appears to function properly when the line is included.

Comment (zurfyx) [reply packages/lexical/src/LexicalUtils.ts:654]: Makes sense, yeah, some E2E tests are flaky and take multiple times to pass unfortunately

**Pull request merged**